packages/expo-file-system/ios/Next/FileSystemNextDirectory.swift

```diff
@@ -0,0 +1,13 @@
+import Foundation
+import ExpoModulesCore
+
+internal final class FileSystemNextDirectory: FileSystemNextPath {
+  func create() throws {
+    try FileManager.default.createDirectory(at: url, withIntermediateDirectories: false)
+  }
+  func validatePath() throws {
+    guard url.isFileURL && url.hasDirectoryPath else {
+      throw Exception(name: "wrong type", description: "tried to create a directory with a file path")
```

Comment:
I know we need to move it to a named exception, but I think I need to coerce this anyhow.